src/abstract-project-create-command.ts

```diff
@@ -140,69 +119,119 @@ export default abstract class AbstractProjectCreateCommand extends AbstractAuthe
     assertPresent({
       authenticator,
       eventSender,
-      commandGenerateConfigGetter,
       projectCreator,
       database,
       messages,
       spinner,
     });
 
     this.eventSender = eventSender;
-    this.commandGenerateConfigGetter = commandGenerateConfigGetter;
     this.projectCreator = projectCreator;
     this.database = database;
     this.messages = messages;
     this.spinner = spinner;
   }
 
+  protected async runAuthenticated(): Promise<void> {
+    try {
+      const { appConfig, dbConfig, meta, authenticationToken } = await this.getConfig();
+
+      this.spinner.start({ text: 'Creating your project on Forest Admin' });
+      const projectCreationPromise = this.projectCreator.create(
+        authenticationToken,
+        appConfig,
+        meta,
+      );
+      const { id, envSecret, authSecret } = await this.spinner.attachToPromise(
+        projectCreationPromise,
+      );
+
+      this.eventSender.meta.projectId = id;
+
+      await this.testDatabaseConnection(dbConfig);
+
+      await this.generateProject({
+        dbConfig,
+        appConfig,
+        forestAuthSecret: authSecret as string,
+        forestEnvSecret: envSecret as string,
+      });
+
+      await this.notifySuccess();
+    } catch (error) {
+      // Display customized error for non-authentication errors.
+      if (error.status !== 401 && error.status !== 403) {
+        this.logger.error(['Cannot generate your project.', `${this.messages.ERROR_UNEXPECTED}`]);
+        this.logger.log(`${this.chalk.red(error)}`);
+        this.exit(1);
+      } else {
+        throw error;
+      }
+    }
+  }
+
+  protected abstract processArguments(programArguments: { [name: string]: any }): Promise<{
```

Comment:
This function name is not self explanatory...

I would rather call it getConfigFormArguments

Reply:
👍